### ISSUE TITLE
[Bug] Can't open PDF files from Box directory in Adobe Acrobat

### ISSUE THREAD

**Issue by rhstanton** [mucommander/mucommander#1268]
### Is there an existing issue for this?

- [X] I have searched the existing issues (including the closed ones)

### Description

If I click on a PDF file in my Box directory, Adobe Acrobat opens but does not show the file. Other applications (e.g., Open With -> Skim) work fine. Note: if I've explicitly asked Box to download the file locally, it's fine. This is a PDF file that has been shared with me but that has not been downloaded locally (note that muCommander shows the file OK in the directory window, and if I ask to copy the file to another directory it works correctly. The problem seems to be solely with opening the file using Adobe Acrobat). Also note that after I run Open With -> Skim, Acrobat works fine in future with that file, presumably because the file actually gets downloaded locally during the Open With -> Skim process.

### Steps to reproduce

Go to Box directory. Click on a PDF file. 

### Expected behavior

Acrobat should open, showing the relevant file

### Actual behavior

Acrobat opens but does not display the file (and the file is not available in some other Acrobat window)

### muCommander version

Version: 1.6.0
Build date: 2024/09/17
Build number: c47a14f

### Java version

Runtime version: 24-ea
VM name: OpenJDK 64-Bit Server VM
VM version: 24-ea+15-1658
VM vendor: Oracle Corporation

### Operating System type and version

Name: Mac OS X
Version: 15.0
Architecture: aarch64

**Comment by rhstanton** [comment]
Quick follow-up: it all works fine when I do the same thing from a Finder window instead of muCommander. I also note that Finder displays a little indicator for whether the file has been downloaded locally or not.

**Comment by rhstanton** [comment]
Yes, I could have been clearer! Yes, it's box.com. I'll do some experiments and try and get some log files.

**Comment by ahadas** [comment]
> Yes, I could have been clearer! Yes, it's box.com. I'll do some experiments and try and get some log files.

sec, I meant that we first need to add some useful logs and then see
@rhstanton until then, could you share a file with my new Box account (my email: arikhadas1 at gmail.com)?

**Comment by rhstanton** [comment]
I don't think it matters what file you use, as long as it's a PDF file and you have Adobe Acrobat installed so that when you double-click on the file in muCommander, Acrobat pops up and (tries to) open the file. You also need to install the macOS Box app on your machine and make sure you're syncing the online directory in which you've put the PDF file. Then open muCommander, navigate to the relevant on-disk directory (probably somewhere under ~/Library/CloudStorage/), double-click on the name of the PDF file, and see what happens.

**Comment by ahadas** [comment]
yeah, I've already installed the Box app on my machine and found my files in `/Users/ahadas/Library/CloudStorage/Box-Box/` but I couldn't reproduce this issue with a PDF that was uploaded through the web browser to the Box drive so I suspected it might be somehow related to the fact that the file was shared with you (one note though - I didn't install Acrobat Reader but used macOS's Preview, @rhstanton can you change the default application that opens PDF to `Preview` and see if the issue still happens?)

**Comment by rhstanton** [comment]
Will do. I think the use of Acrobat is important here. Things worked OK when I (manually) invoked Skim instead.

**Comment by rhstanton** [comment]
Yup. Things work fine when I use Preview as the default application.

**Comment by rhstanton** [comment]
Switching back to Acrobat, the file I just opened using Preview now works fine (presumably because it was actually downloaded when I used Preview), but other PDF files exhibit the same problem as before. So in your experiments, after switching to Acrobat, make sure not to use any files you've previously opened using Preview.

**Comment by rhstanton** [comment]
Here's the debug console log file when the problem occurs (set to "Finest"):

```
[2025-01-01 07:23:07.755] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@a1ed79: setting forceRefresh as MainFrame gained focus

[2025-01-01 07:23:07.755] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@7b26d1ac: setting forceRefresh as MainFrame gained focus

[2025-01-01 07:23:07.990] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-01 07:23:07.990] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-01 07:23:11.138] FINEST FocusDialog.java#windowActivated,220 requesting focus on initial focus component

[2025-01-01 07:23:11.151] FINEST FocusDialog.java#windowActivated,228 requestFocusInWindow failed, calling requestFocus

[2025-01-01 07:23:11.169] FINE FocusRequester.java#run,102 focus requested on javax.swing.JButton

[2025-01-01 07:23:15.279] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@a1ed79: setting forceRefresh as MainFrame gained focus

[2025-01-01 07:23:15.279] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@7b26d1ac: setting forceRefresh as MainFrame gained focus

[2025-01-01 07:23:15.575] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-01 07:23:15.576] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-01 07:23:18.370] FINE AbstractProcess.java#startMonitoring,90 Starting process merged output monitor...

[2025-01-01 07:23:18.493] FINE ProcessOutputMonitor.java#run,115 Process output stream emptied, closing

[2025-01-01 07:23:25.297] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@a1ed79: setting forceRefresh as MainFrame gained focus

[2025-01-01 07:23:25.297] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@7b26d1ac: setting forceRefresh as MainFrame gained focus

[2025-01-01 07:23:25.598] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-01 07:23:25.598] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

```

**Comment by rhstanton** [comment]
This is what I see:
```
[2025-01-19 09:46:28.794] FINE LocalProcess.java#<init>,57 Starting local process, tokens: [open, /Users/stanton/Box/scratch/test.pdf], dir: /Users/stanton/Box/scratch

[2025-01-19 09:46:28.808] FINE AbstractProcess.java#startMonitoring,90 Starting process merged output monitor...

[2025-01-19 09:46:28.945] FINE ProcessOutputMonitor.java#run,115 Process output stream emptied, closing

[2025-01-19 09:46:36.145] FINE FolderChangeMonitor.java#windowGainedFocus,344 com.mucommander.core.FolderChangeMonitor@271e1ba8: setting forceRefresh as MainFrame gained focus

[2025-01-19 09:46:36.167] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-19 09:46:36.167] FINE ModificationDateBasedMonitoredFile.java#isChanged,42 isChanged = false

[2025-01-19 09:46:39.379] FINEST FocusDialog.java#windowActivated,220 requesting focus on initial focus component

[2025-01-19 09:46:39.387] FINEST FocusDialog.java#windowActivated,228 requestFocusInWindow failed, calling requestFocus

[2025-01-19 09:46:39.399] FINE FocusRequester.java#run,102 focus requested on javax.swing.JButton

```

**Comment by rhstanton** [comment]
This prompted me to run
```
open test.pdf
```
at the command line - same problem.

**Comment by rhstanton** [comment]
And I see others have noted a similar issue. E.g., https://support.box.com/hc/en-us/community/posts/27461317975827-Box-Drive-on-Mac-won-t-open-PDF-with-acobat-set-as-default-unless-I-make-file-available-offline

So this does not seem to be a muCommander issue. Thanks for checking into it!
